pkg/certificate/manager.go

```diff
@@ -41,42 +42,47 @@ func NewManager(mrcClient MRCClient, serviceCertValidityDuration time.Duration,
 func (m *Manager) Start(checkInterval time.Duration, stop <-chan struct{}) {
 	ticker := time.NewTicker(checkInterval)
 	go func() {
-		m.checkAndRotate()
+		m.checkAndRotateServiceCerts()
 		for {
 			select {
 			case <-stop:
 				ticker.Stop()
 				return
 			case <-ticker.C:
-				m.checkAndRotate()
+				m.checkAndRotateServiceCerts()
 			}
 		}
 	}()
 }
 
-func (m *Manager) checkAndRotate() {
+func (m *Manager) checkAndRotateServiceCerts() {
 	// NOTE: checkAndRotate can reintroduce a certificate that has been released, thereby creating an unbounded cache.
 	// A certificate can also have been rotated already, leaving the list of issued certs stale, and we re-rotate.
 	// the latter is not a bug, but a source of inefficiency.
 	for _, cert := range m.ListIssuedCertificates() {
-		shouldRotate := cert.ShouldRotate()
+		if cert.GetCertificateType() != Service {
+			log.Debug().Msgf("Skipping check and rotate of cert %s of type %s since it is not of type Service",
+				cert.GetCommonName(),
+				cert.GetCertificateType())
+			continue
+		}
+
+		newCert, rotated, err := m.IssueCertificate(cert.GetCommonName(), m.serviceCertValidityDuration(), Service)
```

Comment:
Looks like the rotated bool is only used here. Instead of modifying the API and exposing that info when callers don't seem to care, what if we did:

```go
rotated := cert != newCert
```

And kept the old method signature